### PR TITLE
recreate BackedEnums when applying Manipulations

### PR DESCRIPTION
**Pull request by tkempf**
Fixes Issue #3582

### CODE REVIEW

File: src/Conversions/Manipulations.php [hunk 1]
```diff
@@ -52,6 +60,34 @@ public function isEmpty(): bool
     public function apply(ImageDriver $image): void
     {
         foreach ($this->manipulations as $manipulationName => $parameters) {
+            switch($manipulationName){
```
Comment (Nielsvanpach) [inline src/Conversions/Manipulations.php:63]: Could you use a `match` operator instead?

File: src/Conversions/Manipulations.php [hunk 2]
```diff
@@ -52,6 +60,34 @@ public function isEmpty(): bool
...
+                case 'border':
+                    (isset($parameters['type'])) && $parameters['type'] = BorderType::from($parameters['type']) ;
```
Comment (Nielsvanpach) [inline src/Conversions/Manipulations.php:65]:
```suggestion
                    (isset($parameters['type'])) && $parameters['type'] = BorderType::from($parameters['type']);
```

**Review by Nielsvanpach — COMMENTED** [review]
Could you add a test case to make sure this fix solves the problem?

File: src/Conversions/Manipulations.php (hunk 1 above)
Comment (tkempf) [reply src/Conversions/Manipulations.php:63]: What do you think about using neither match nor switch. Just recreate all Enums if the parameter is present independent of thy type of manipulation
```suggestion
    (isset($parameters['type'])) && $parameters['type'] = BorderType::from($parameters['type']) ;
    (isset($parameters['fit'])) && $parameters['fit'] = Fit::from($parameters['fit']);
    (isset($parameters['position'])) && $parameters['position'] = AlignPosition::from($parameters['position']);
    (isset($parameters['colorFormat'])) && $parameters['colorFormat'] = ColorFormat::from($parameters['colorFormat']);
    (isset($parameters['constraints'])) && $parameters['constraints'] = Constraint::from($parameters['constraints']);
    (isset($parameters['position'])) && $parameters['position'] = CropPosition::from($parameters['position']);
    (isset($parameters['fit'])) && $parameters['fit'] = Fit::from($parameters['fit']);
    (isset($parameters['flip'])) && $parameters['flip'] = FlipDirection::from($parameters['flip']);
```

Comment (freekmurze) [reply src/Conversions/Manipulations.php]: I think a match we be more clear in this case

**Comment by MrMeshok** [comment]
There is also valid usage without named parameters
```php
$media->manipulations = [
    'default' => [
        'border' => [
            1,
            BorderType::Expand,
            '000000'
        ]
    ]
];
```
I see 2 solutions there:
1. Try to find enum from parameters array by named parameter key or numeric position e.g. `BorderType::from($parameters['type'] ?? $parameters[1])`
2. Use reflection and check ImageDriver method parameters before calling it

Obviously reflection would be better, but there performance concerns, so it needs benchmarking.

I could take over this issue as this also need for my use case

**Comment by timvandijck** [comment]
@tkempf thank you for your work!

@MrMeshok I'm going to already merge this PR, but if you have some more improvements on this topic I would be happy to look at a new PR from you.

**Pull request merged**